src/uct/ib/dc/accel/dc_mlx5.c

```diff
@@ -1087,6 +1087,163 @@ static void uct_dc_mlx5_iface_event_cq(uct_ib_iface_t *ib_iface,
     iface->mlx5_common.cq[dir].cq_sn++;
 }
 
+#if HAVE_DC_DV
+static ucs_status_t uct_dc_mlx5_iface_create_qp(uct_ib_iface_t *iface,
+                                                uct_ib_qp_attr_t *attr,
+                                                struct ibv_qp **qp_p)
+{
+    uct_ib_device_t *dev               = uct_ib_iface_device(iface);
+    struct mlx5dv_qp_init_attr dv_attr = {};
+    struct ibv_qp *qp;
+
+    uct_ib_iface_fill_attr(iface, attr);
+    attr->ibv.cap.max_recv_sge          = 0;
+
+    dv_attr.comp_mask                   = MLX5DV_QP_INIT_ATTR_MASK_DC;
+    dv_attr.dc_init_attr.dc_type        = MLX5DV_DCTYPE_DCI;
+    dv_attr.dc_init_attr.dct_access_key = UCT_IB_KEY;
+    qp = mlx5dv_create_qp(dev->ibv_context, &attr->ibv, &dv_attr);
+
+    if (qp == NULL) {
+        ucs_error("iface=%p: failed to create DCI: %m", iface);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    attr->cap = attr->ibv.cap;
+    *qp_p = qp;
+
+    return UCS_OK;
+}
+
+ucs_status_t uct_dc_iface_dci_connect(uct_dc_iface_t *iface,
```

Comment:
is this "overriding" the one from dc_iface.c? not nice, better use iface->ops
name it "uct_dc_mlx5_iface_dci_connect"

Reply:
Dynamic override for uct_dc_iface_create_dct/uct_dc_iface_dci_connect is useless or may it coexists?

Reply:
in theory it may coexist, but mostly it's wrong namespace / module sepraration

Reply:
Current theory says no - uct_dc_iface.rx.dct may be either ibv_exp_dct or ibv_qp at compile time.
I started to extend iface->ops, than noticed that design makes it meaningless and removed.

Reply:
so why is this in dc_mlx5.c? let's move it to dc_iface.c then..

Reply:
ic. can we move uct_dc_iface.rx.dct when it's defined as ibv_qp to dc_mlx5.h?